### PR TITLE
[client/catapult] fix: catapult pylint failure

### PR DESCRIPTION
**Pull request by Wayonb**
## What's the issue?
With Pylint 2.16 catapult lint is failing with ``[W0719(broad-exception-raised), require_non_existence] Raising too general exception: Exception``.
This is due to pylint adds a new checker for broad-exception-raised - ``https://github.com/PyCQA/pylint/issues/7494``

## How have you changed the behavior?
Switch from Exception to RuntimeError 

## How was this change tested?
Run the linting on dev box.

### CODE REVIEW

**Review by Jaguar0625 — APPROVED** [review]

File: sdk/python/symbolchain/sc/__init__.py [hunk 1]
```diff
@@ -14976,7 +14976,7 @@ def deserialize(cls, payload: bytes) -> Block:
 			(NormalBlockV1.BLOCK_TYPE): NormalBlockV1,
 			(ImportanceBlockV1.BLOCK_TYPE): ImportanceBlockV1
 		}
-		discriminator = (parent.type_)
+		discriminator = parent.type_
```
Comment (Jaguar0625) [inline sdk/python/symbolchain/sc/__init__.py:14979]: this is generated code, so need to update generator too ;)

Comment (Wayonb) [reply sdk/python/symbolchain/sc/__init__.py]: I just disable the rule since it is generated code.  Let me know if that's ok.

File: sdk/python/tests/test_QrStorage.py [hunk 2]
```diff
@@ -49,7 +47,7 @@ def test_cannot_save_more_than_max_buffer(self):
 			buffer = TestUtils.randbytes(MAX_DATA_BYTES + 1)
 
 			# Act + Assert:
-			with self.assertRaises(qrcode.exceptions.DataOverflowError):
+			with self.assertRaises(ValueError):
```
Comment (Jaguar0625) [inline sdk/python/tests/test_QrStorage.py:50]: why this change - did exception change?

Comment (Wayonb) [reply sdk/python/tests/test_QrStorage.py:50]: Yea, in v7.4 - https://github.com/lincolnloop/python-qrcode/blob/b80fea6ee7e75f3024b9ed7adf891a143e0b14e3/qrcode/util.py#L183
